webapp/publisher/views.py

```diff
@@ -78,16 +78,53 @@ def get_account_details():
         return _handle_errors(api_error)
 
     flask_user = flask.session["openid"]
+
+    subscriptions = None
+    subscribed_to_newsletter = None
+    # don't rely on marketo to show the page,
+    # if anything fails, just continue and don't show
+    # this section
+    try:
+        marketo = marketo_api.MarketoApi()
+        marketo_user = marketo.get_user(flask_user["email"])
+        marketo_subscribed = marketo.get_newsletter_subscription(
+            marketo_user["id"]
+        )
+        subscribed_to_newsletter = False
+        if (
+            "snapcraftnewsletter" in marketo_subscribed
+            and marketo_subscribed["snapcraftnewsletter"]
+        ):
+            subscribed_to_newsletter = True
+    except Exception:
+        pass
+
+    if subscribed_to_newsletter is not None:
+        subscriptions = {"newsletter": subscribed_to_newsletter}
+
     context = {
         "image": flask_user["image"],
         "username": flask_user["nickname"],
         "displayname": flask_user["fullname"],
         "email": flask_user["email"],
+        "subscriptions": subscriptions,
     }
 
     return flask.render_template("publisher/account-details.html", **context)
 
 
+@account.route("/details", methods=["POST"])
```

Comment:
@tbille @jpmartinspt I'd appreciate some help testing this route. I've tried many things and wasted a lot of time, in the end I've just removed the tests.

Reply:
I do see a test for it, anything in particular you wanted help with?